api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -179,9 +179,9 @@ class GrpcCallHandlerBase {
 
   void cancel();
 
-  virtual void onCreateInitialMetadata() {}
-  virtual void onSuccess(std::unique_ptr<WasmData> message) = 0;
-  virtual void onFailure(GrpcStatus status, std::unique_ptr<WasmData> error_message) = 0;
+  virtual void onCreateInitialMetadata(uint32_t /* headers */) {}
+  virtual void onSuccess(size_t body_size) = 0;
+  virtual void onFailure(GrpcStatus status) = 0;
```

Comment:
Should this include `status_message_size`?

Reply:
We could.  My feeling however is that this is an exceptional situation and if you have an error and need to the details, then an extra call will not kill you.